addon/mixins/geojson-filter-parser.js

```diff
@@ -75,12 +75,19 @@ export default Ember.Mixin.create({
   */
   parseFilter(filter, isInnerExpression) {
     let result = this._super(...arguments);
-    return Ember.isNone(result) || isInnerExpression ? result : `function(feature) { return ${result}; }`;
+    if (isInnerExpression) {
+      return result;
+    }
+
+    if (Ember.isBlank(result)) {
```

Comment:
isNone was intentional. Because an empty string is returned for empty filter, null/undefined for not valid filter.